docs/build/quickstarts.md

```diff
@@ -1,3 +1,200 @@
+
+## What are deals
+
+Deal represents a request from Developer to host a set of WebAssembly functions for a specified price. Providers are able to join these deals in order to host functions and receive tokens for that.
+
+In essense, Deal specifies what to deploy, and how much to pay for that.
+
+Among parameters described above, each Deal specifies an `AppCID`. `AppCID` is an IPFS CID that points to the `Worker Definition` data structure stored on IPFS.
+
+
+### Where `Worker Definition` comes from
+
+As we will see in `step-by-step` section below, there's a certain structure to the Fluence Projects. That structure is defined by `fluence cli`, which provides means to initialize and maintain Fluence Projects.
+
+Fluence Project can hold any number of Rust and WebAssembly -based functions and timer-based recurrent Aqua scripts.
+
+A set of functions defined by a single Rust project or a set of linked WebAssembly modules is called a `service`. You can read more about services [HERE](https://www.youtube.com/watch?v=tgTUtfb0Ok8).
+
+A timer-based recurrent Aqua scripts are called Spells. It's like a cron, but for distributed choreography. You can read more about spells [HERE](https://www.youtube.com/watch?v=tgTUtfb0Ok8).
+
+Developers can group Services and Spells to Deals, and then deploy each deal via `fluence deal deploy` command in `fluence cli`. Before Deal is registered on Chain, its settings and artifacts are uploaded to IPFS to produce `Worker Definition`, which is again stored on IPFS to produce `AppCID`.
+
+### How deals are matched
+
+In order for a Deal to be deployed, there should be Providers willing to host it. That process is governed by the `Matcher` smart contract.
+
+It's important that Matcher makes it possible for a single Deal to be deployed on several hosts to provide High Availability.
+
+A Deal specifies different parameters, like `price per epoch`, `minimum collateral`, `effectors` and `target number of workers`.
+
+These parameters are matched against Providers' Market Offers, which specify `minimum price per epoch`, `maximum collateral` and `available effectors`.
+
+You can read more about effectors [HERE](https://www.youtube.com/watch?v=tgTUtfb0Ok8).
+
+Providers register their Market Offers on chain in a contract called `Matcher`, and each `deal deploy` calls that contract to find Offers compatible with that Deal. It should match the price, collateral and available `effectors`.
+
+Each match should find at least `target number of workers` Providers whose offers are compatible with a given Deal.
+
+After match happened, Providers recieve an event from Chain that commands them to deploy Deal to their chosen Compute Peer.
+
+You can read a more detailed description of Matching process [HERE](https://www.youtube.com/watch?v=tgTUtfb0Ok8).
+
+## What are workers
+
+When Deal is matched against Market Offers, it gets deployed to a number of Compute Peers.
```

Comment:
Maybe a brief note on what is Compute Peer would be great somewhere near here. It can be deduced from context, but just to be more clear

Reply:
nice rick roll.